File: ca-reservation/core/src/main/java/io/sharing/server/core/support/stereotype/UseCase.java
package io.sharing.server.core.support.stereotype;

import org.springframework.stereotype.Component;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.*;

@Retention(RUNTIME)
@Target(TYPE)
@Component
public @interface UseCase {
}

File: ca-product/core/src/main/kotlin/io/sharing/server/product/core/support/stereotype/UseCase.kt
package io.sharing.server.core.support.stereotype

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Component

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Component
annotation class UseCase(
    @get:AliasFor(annotation = Component::class)
    val value: String = ""
)
